src/Controller/Exceptions/ProdutoJaAdicionadoAoCarrinhoException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Controller\Exceptions;
+
+class  ProdutoJaAdicionadoAoCarrinhoException extends \Exception
+{
+    public function __construct()
+    {
+        parent::__construct("Produto ja adicionado ao carrinho .");
+    }
+} 
```

Comment:
Faz um commit changes daqui, só ajustandos espaços e tals

```suggestion
class ProdutoJaAdicionadoAoCarrinhoException extends \Exception
{
    public function __construct()
    {
        parent::__construct('Produto ja adicionado ao carrinho.');
    }
} 

```

Reply:
se ficar o  commit suggestions tu precisa fazer um pull na branch local